### PR TITLE
ci(tests): use 'concurrency' instead of 'styfle/cancel-workflow-action'

### PR DESCRIPTION
**Pull request by gurza**
Solve #2590

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -15,9 +19,6 @@ permissions:
 
 jobs:
   test:
-    permissions:
-      actions: write  # for styfle/cancel-workflow-action to cancel/stop running workflows
-      contents: read  # for actions/checkout to fetch code
```
Comment (gmlewis) [inline .github/workflows/tests.yml:20]: I thought that this `contents: read` permission was orthogonal to the `actions:` permission and ensured that bad actors cannot write to the repo within an action.

If this is the case, I think we should keep this permission and not delete it.

Thoughts?

Comment (gurza) [reply .github/workflows/tests.yml:20]: @gmlewis thank you for your check.

In `tests.yml` there is high level `permissions` section which is set to read access.

```
permissions:
  contents: read
```

I have deleted `jobs/test/permissions` subsection I think it enhances read access. As you can see in the comment `actions: write` is used for `styfle/cancel-workflow-action`.

Or am I missing something?

Comment (gmlewis) [reply .github/workflows/tests.yml:20]: Ah, so that `contents: read` was redundant!  OK, thank you, @gurza !

**Review by gmlewis — APPROVED** [review]
Thank you, @gurza !
LGTM.

Awaiting signed CLA before merging.

**Pull request merged**